src/drivers/rpm_capture/RPMCapture.cpp

```diff
@@ -0,0 +1,220 @@
+/****************************************************************************
+ *
+ *   Copyright (c) 2021 PX4 Development Team. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+
+#include "RPMCapture.hpp"
+#include <px4_arch/io_timer.h>
+#include <board_config.h>
+#include <parameters/param.h>
+#include <px4_platform_common/events.h>
+#include <systemlib/mavlink_log.h>
+
+RPMCapture::RPMCapture() :
+	ScheduledWorkItem(MODULE_NAME, px4::wq_configurations::hp_default),
+	ModuleParams(nullptr)
+{
+	_pwm_input_pub.advertise();
+	ScheduleNow();
+}
+
+RPMCapture::~RPMCapture()
+{
+	if (_channel >= 0) {
+		io_timer_unallocate_channel(_channel);
+		px4_arch_gpiosetevent(_rpm_capture_gpio, false, false, false, nullptr, nullptr);
+	}
+}
+
+bool RPMCapture::init()
+{
+	bool success = false;
+
+	for (unsigned i = 0; i < 16; ++i) {
+		char param_name[17];
+		snprintf(param_name, sizeof(param_name), "%s_%s%d", PARAM_PREFIX, "FUNC", i + 1);
+		param_t function_handle = param_find(param_name);
+		int32_t function;
+
+		if (function_handle != PARAM_INVALID && param_get(function_handle, &function) == 0) {
+			if (function == 2070) { // RPM_Input
```

Comment:
why 2070?

Reply:
It's defined as a function in this pr:
https://github.com/PX4/PX4-Autopilot/blob/bdec521b0a629d47d053a31d303f41913e355dc8/src/lib/mixer_module/output_functions.yaml#L64
I'm not sure if that's defined outside of the metadata 👀 otherwise I'll extend the comment.

Reply:
a5f81a19f3f33c50aea23722798cb1ba76cf8a12